scripts/update-plugin-list.py

```diff
@@ -0,0 +1,85 @@
+import datetime
+import pathlib
+import re
+
+import packaging.version
+import requests
+import tabulate
+
+FILE_HEAD = r"""Plugins List
+============
+
+PyPI projects that match "pytest-\*" are considered plugins and are listed
+automatically. Packages classified as inactive are excluded.
+"""
+DEVELOPMENT_STATUS_CLASSIFIERS = (
+    "Development Status :: 1 - Planning",
+    "Development Status :: 2 - Pre-Alpha",
+    "Development Status :: 3 - Alpha",
+    "Development Status :: 4 - Beta",
+    "Development Status :: 5 - Production/Stable",
+    "Development Status :: 6 - Mature",
+    "Development Status :: 7 - Inactive",
+)
+
+
+def iter_plugins():
+    regex = r">([\d\w-]*)</a>"
+    response = requests.get("https://pypi.org/simple")
+    for match in re.finditer(regex, response.text):
+        name = match.groups()[0]
+        if not name.startswith("pytest-"):
+            continue
+        response = requests.get(f"https://pypi.org/pypi/{name}/json")
+        if not response.ok:
```

Comment:
If we fail to fetch, I think it is better to fail (or retry and fail) instead of silently skipping.

Reply:
The first case in the list that will be an issue will be pytest-azurepipelines42. [This](https://pypi.org/project/pytest-azurepipelines42) would be the project page and [this](https://pypi.org/pypi/pytest-azurepipelines42/json) is the place for its JSON API. I do not know why they are empty, but here is a complete list of the current cases which pass the condition with the link to the API:
[pytest-azurepipelines42](https://pypi.org/pypi/pytest-azurepipelines42/json), [pytest-core](https://pypi.org/pypi/pytest-core/json), [pytest-locust](https://pypi.org/pypi/pytest-locust/json), [pytest-locustio](https://pypi.org/pypi/pytest-locustio/json), [pytest-regression](https://pypi.org/pypi/pytest-regression/json), [pytest-runtime-info](https://pypi.org/pypi/pytest-runtime-info/json), [pytest-testraill](https://pypi.org/pypi/pytest-testraill/json)

I would prefer not to have a blacklist. Maybe there will be working available packages in the future. This is not really failing to fetch, since there actually is nothing to fetch. I understand, that this kind of handling may cause problems, but i don't know how exactly, since i believe connectivity issues like timeouts will throw an error in the row before.

Reply:
Really strange that some projects are listed but return 404.

Agree about the blacklist, we shouldn't do that.

What I suggest is something like this:

```py
response = requests.get(f"https://pypi.org/pypi/{name}/json")
if response.status_code == 404:
    # Some packages, like pytest-azurepipelines42, are included in https://pypi.org/simple but
    # return 404 on the JSON API. Skip.
    continue
response.raise_for_status()
```